unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -1348,6 +1349,50 @@ notifyUser dir o = case o of
         )
       p = prettyShortHash . SH.take hqLength
       c = P.syntaxToColor
+  ListNamespaceDependencies _ppe Empty Empty -> pure $ "This namespace has no external dependencies."
+  ListNamespaceDependencies ppe externalTerms externalTypes ->
+    pure . P.lines $
+      Monoid.whenM
+        (not . null $ externalTerms)
+        [ P.bold "This namespace depends on the following external terms:"
+            <> P.newline
+            <> P.indent "  " (prettyTerms externalTerms)
+        ]
+        <> Monoid.whenM
+          (not . null $ externalTypes)
+          [ P.newline <> P.bold "This namespace depends on the following external types declarations and constructors:"
+              <> P.newline
+              <> P.indent "  " (prettyTypes externalTypes)
+          ]
+    where
+      prettyTerms :: Map Reference (Type v Ann) -> P.Pretty P.ColorText
+      prettyTerms m =
+        m
+          & Map.toList
+          & fmap (first Referent.fromReference)
+          & fmap (\(r, typ) -> TypePrinter.prettySignaturesCTCollapsed ppe [(r, PPE.typeOrTermName ppe r, typ)])
+          & P.lines
+
+      prettyTypes :: Map Reference (Set Referent) -> P.Pretty P.ColorText
+      prettyTypes m =
+        m
+          & Map.toList
+          & fmap (first Referent.fromReference)
+          & fmap
+            ( \(r, constructors) ->
+                P.text (HQ.toText $ PPE.typeOrTermName ppe r)
+                  <> Monoid.whenM
+                    (not . null $ constructors)
+                    ( P.newline
+                        <> ( P.indent (P.hiBlack "  constructor ")
+                               . P.lines
+                               . fmap (P.text . HQ.toText . PPE.typeOrTermName ppe)
+                               . Set.toList
+                               $ constructors
+                           )
+                    )
+            )
+          & P.lines
```

Comment:
The combination of `&` and `.` and `$` in the same expression is exciting.

Reply:
Yeah I'll pull out the constructor printer into a separate func.